benches/functional.rs

```diff
@@ -0,0 +1,101 @@
+use bst::functional::Tree;
+use criterion::{
+    black_box, criterion_group, criterion_main, AxisScale, BenchmarkId, Criterion,
+    PlotConfiguration,
+};
+
+/// Returns how many nodes are needed to fill a binary tree with `num_levels` levels.
+fn num_nodes_in_full_tree(num_levels: usize) -> usize {
+    2usize.pow(num_levels as u32) - 1
```

Comment:
I don't think I understand the `2usize` syntax.  Is that like `2 as usize`?

Reply:
Good question - `as usize` means "take this thing that isn't a `usize` and cast it to `usize`". Specifically "`as`" is a "primitive cast". In this case we are saying "here is a `usize` with value `2`" so it's not taking a `2i32` and making it a `usize` - it's declaring a new `usize` with the value `2`. And this is only needed because `pow` is defined for all sorts of numbers and Rust doesn't know which one to pick for `2.pow` (not sure why it doesn't assume the default of `i32` here :shrug:)